module/surveySubmissions/helper.js

```diff
@@ -0,0 +1,424 @@
+ * author : Deepa
+ * created-date : 07-Spe-2020
+ * Description : Survey submissions helper functionality.
+ */
+
+// Dependencies
+const submissionsHelper = require(MODULES_BASE_PATH + "/submissions/helper");
+const kafkaClient = require(ROOT_PATH + "/generics/helpers/kafkaCommunications");
+const slackClient = require(ROOT_PATH + "/generics/helpers/slackCommunications");
+
+/**
+    * SurveySubmissionsHelper
+    * @class
+*/
+module.exports = class SurveySubmissionsHelper {
+
+
+   /**
+   * find survey submissions
+   * @method
+   * @name surveySubmissionDocuments
+   * @param {Array} [surveySubmissionFilter = "all"] - survey submission ids.
+   * @param {Array} [fieldsArray = "all"] - projected fields.
+   * @param {Array} [sortedData = "all"] - sorted field.
+   * @param {Array} [skipFields = "none"] - field not to include
+   * @returns {Array} List of survey submissions. 
+   */
+  
+  static surveySubmissionDocuments(
+    surveySubmissionFilter = "all", 
+    fieldsArray = "all",
+    sortedData = "all",
+    skipFields = "none"
+  ) {
+    return new Promise(async (resolve, reject) => {
+        try {
+    
+            let queryObject = (surveySubmissionFilter != "all") ? surveySubmissionFilter : {};
+    
+            let projection = {}
+    
+            if (fieldsArray != "all") {
+                fieldsArray.forEach(field => {
+                    projection[field] = 1;
+                });
+            }
+
+            if( skipFields !== "none" ) {
+              skipFields.forEach(field=>{
+                projection[field] = 0;
+              })
+            }
+
+            let surveySubmissionDocuments;
+
+            if ( sortedData !== "all" ) {
+                
+                surveySubmissionDocuments = 
+                await database.models.surveySubmissions.find(
+                    queryObject, 
+                    projection
+                ).sort(sortedData).lean();
+
+            } else {
+                
+                surveySubmissionDocuments = 
+                await database.models.surveySubmissions.find(
+                    queryObject, 
+                    projection
+                ).lean();
+            }   
+
+            return resolve(surveySubmissionDocuments);
+            
+        } catch (error) {
+            return resolve({
+                success: false,
+                message: error.message,
+                data: false
+            })
+        }
+    });
+ }
+
+
+
+    /**
+    * Make survey submission.
+    * @method
+    * @name make
+    * @param {Object} req - request details 
+    * @returns {String} -  message.
+    */
+
+   static make( req = {} ) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if (Object.keys(req).length == 0) {
+                throw new Error(messageConstants.apiResponses.REQUEST_DETAILS_REQUIRED)
+            }
+
+            let surveySubmissionDocument = await this.surveySubmissionDocuments
```

Comment:
@deepakr123 - Put this condition in isAllowed.

Reply:
resolved